yaml/github-actions/rubygems-publish-key.yaml

```diff
@@ -0,0 +1,20 @@
+rules:
+  - id: rubygems-publish-key
+    message: |
+      Found long-term access key. Instead prefer RubyGems trusted publishing and
+      temporary OIDC security credentials.
+    languages: [yaml]
+    severity: WARNING
+    metadata:
+      category: security
+      cwe: "CWE-798: Use of Hard-coded Credentials"
+      subcategory: [audit]
+      confidence: HIGH
+      likelihood: HIGH
+      impact: MEDIUM
+      technology: [rubygems, github-actions]
+      references:
+        - https://guides.rubygems.org/trusted-publishing/
+        - https://guides.rubygems.org/api-key-scopes/#usage-with-gem-cli
+        - https://github.com/rubygems/release-gem
+    pattern: GEM_HOST_API_KEY
```

Comment:
Should we match `uses: ruby/setup-ruby` to reduce false positives?

Reply:
I think this is just restricted to `gem push`, so FPs should be low. My concern with limiting to `uses: ruby/setup-ruby` is that `GEM_HOST_API_KEY` could occur in many places in the GHA file. The `env` can be set globally, per-job, per-step, etc.

Reply:
Dropping the `likelihood` to `LOW` though 👍